Sources/TuistKit/Generator/GeneratorModelLoader.swift

```diff
@@ -107,6 +102,42 @@ class GeneratorModelLoader: GeneratorModelLoading {
             return locateDirectoryTraversingParents(from: from.parentDirectory, path: path, fileHandler: fileHandler)
         }
     }
+
+    private func enriched(model: TuistGenerator.Project,
+                          with config: TuistGenerator.TuistConfig) throws -> TuistGenerator.Project {
+        var enrichedModel = model
```

Comment:
Notice that the model being passed is a reference, so creating another internal variable does not make any copy. I'd just remove it.

Reply:
We do the following operations in the same method:
```swift
enrichedModel = enrichedModel.adding(target: manifestTarget)
...
enrichedModel = enrichedModel.replacing(fileName: xcodeFileName)
```
The idea was to be consistent and continue returning a new model every time a `Project` property is changed. `Project`'s properties are immutable. 

The alternative would be to change the `let`s inside `Project` to `var`s.

Reply:
We do the following operations in the same method:
```swift
enrichedModel = enrichedModel.adding(target: manifestTarget)
...
enrichedModel = enrichedModel.replacing(fileName: xcodeFileName)
```
The idea was to be consistent and continue returning a new model every time a `Project` property is changed. `Project`'s properties are immutable. 

The alternative would be to change the `let`s inside `Project` to `var`s.

Reply:
Let's keep it like that for now. If we see maintaining those methods that create copies of the instances become cumbersome, we can reconsider the approach.